velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -626,6 +655,37 @@ void PageReader::makeDecoder() {
               type_->typeLength_,
               true);
           break;
+        case thrift::Type::INT64:
+          if (type_->logicalType_.has_value() &&
+              type_->logicalType_.value().__isset.TIMESTAMP) {
+            auto logicalType = type_->logicalType_.value();
+
+            VELOX_CHECK(
+                logicalType.TIMESTAMP.isAdjustedToUTC,
+                "Only UTC adjusted Timestamp is supported.");
+            VELOX_CHECK(
+                !logicalType.TIMESTAMP.unit.__isset.NANOS,
+                "Nano Timestamp unit not supported.");
+
+            auto precisionUnit = logicalType.TIMESTAMP.unit.__isset.MICROS
+                ? dwio::common::TimestampPrecision::kMicros
+                : dwio::common::TimestampPrecision::kMillis;
```

Comment:
Do we need to check `__isset.MILLIS` before assign kMillis, and throw for the other units?

Reply:
There are only 3 units. For Nano we throw error a line above